ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -608,6 +716,7 @@ public static short getVersion(String idStr) throws IOException {
     BlobIdPreamble(DataInputStream stream) throws IOException {
       version = stream.readShort();
       byte blobIdFlag;
+      blobDataType = null;
```

Comment:
set this to null in the individual cases

Reply:
done